### PR TITLE
Add group history endpoint

### PR DESCRIPTION
**Pull request by cl8n**
part 3 ~
- [x] depends on #9487

will be used with a view for the group history page, and as json for a "show more" or different search action on that page.

### CODE REVIEW

File: app/Transformers/UserGroupEventTransformer.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        unset($json['actor_name']);
+
+        return $json;
+    }
+
+    public function includeActor(UserGroupEvent $event): ResourceInterface
+    {
+        return $this->primitive([
+            'id' => $event->actor_id,
+            'name' => $event->details['actor_name'],
```
Comment (nanaya) [inline app/Transformers/UserGroupEventTransformer.php:48]: username?

**Review by nanaya — CHANGES_REQUESTED** [review]

**Comment by peppy** [comment]
@cl8n wanna apply the reviews here to keep things moving? :)

File: app/Http/Controllers/GroupHistoryController.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        $query = UserGroupEvent::visibleForUser(auth()->user());
+        $skipQuery = false;
+
+        if ($params['after'] !== null) {
+            $query->where('created_at', '>', $params['after']);
+        }
+
+        if ($params['before'] !== null) {
+            $query->where('created_at', '<', $params['before']);
+        }
```
[the remark this answers is not in the document]
Comment (cl8n) [reply app/Http/Controllers/GroupHistoryController.php]: renamed to min_date max_date , these names are used in beatmapset event search

File: app/Http/Controllers/GroupHistoryController.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+        if ($params['group'] !== null) {
+            // Not `app('groups')->byIdentifier(...)` because that would create the group if not found
+            $groupId = app('groups')->allByIdentifier()->get($params['group'])?->getKey();
+
+            if ($groupId !== null) {
+                $query->where('group_id', $groupId);
+            } else {
+                $skipQuery = true;
```
[the remark this answers is not in the document]
Comment (cl8n) [reply app/Http/Controllers/GroupHistoryController.php]: scoped to none() the code was getting really messy lol

**Comment by peppy** [comment]
@ppy/team-web can you check on this one? looks like it's ready for review.

**Review by nanaya — APPROVED** [review]

**Pull request merged**